force_wfmanager/wfmanager_global_task.py

```diff
@@ -0,0 +1,147 @@
+import logging
+
+from pyface.tasks.action.api import SMenu, SToolBar, TaskAction
+from pyface.api import ImageResource
+from traits.api import List
+
+from force_wfmanager.wfmanager import TaskToggleGroupAccelerator
+
+from traits.api import Callable
+from envisage.ui.tasks.task_extension import TaskExtension
+from pyface.tasks.action.api import SchemaAddition
+
+log = logging.getLogger(__name__)
+
+
+class WfManagerGlobalTask(TaskExtension):
+    """Global extensions to Setup and Review tasks.
+    """
+
+    # The ID of the task to extend. If the ID is omitted, the extension applies
+    # to all tasks.
+    # NO TASK_ID as this is a global task extension
+
+    # A list of menu bar and tool bar items to add to the set provided
+    # by the task.
+    actions = List(SchemaAddition)
+
+    # A list of dock pane factories that will extend the dock panes provided by
+    # the task.
+    dock_pane_factories = List(Callable)
+
+    def _actions_default(self):
+        """ Traits default.
+        :return: list of schema's to be added to all tasks ("globally").
+        """
+
+        return [
+                SchemaAddition(
+                    path='mymenu',
+                    factory=self.file_menu,
+                    id='file.menu.schema',
+                ),
+                SchemaAddition(
+                    path='mymenu',
+                    factory=self.view_menu,
+                    id='file.view.schema',
+                ),
+                SchemaAddition(
+                    path='mymenu',
+                    factory=self.help_menu,
+                    id='file.help.schema',
+                ),
+                SchemaAddition(
+                    path='',
+                    factory=self.run_tools,
+                    id='tools.run.schema',
+                ),
+            ]
+
+    def file_menu(self):
+        """ File menu schema.
+        """
+        return SMenu(
+                TaskAction(
+                    name="Open Workflow...",
+                    method="setup_task.open_workflow",
+                    enabled_name="setup_task.save_load_enabled",
+                    accelerator="Ctrl+O",
+                ),
+                TaskAction(
+                    id="Save",
+                    name="Save Workflow",
+                    method="setup_task.save_workflow",
+                    enabled_name="setup_task.save_load_enabled",
+                    accelerator="Ctrl+S",
+                ),
+                TaskAction(
+                    name="Save Workflow as...",
+                    method="setup_task.save_workflow_as",
+                    enabled_name="setup_task.save_load_enabled",
+                    accelerator="Shift+Ctrl+S",
+                ),
+                TaskAction(name="Plugins...",
+                           method="setup_task.open_plugins"),
+                TaskAction(name="Exit", method="exit"),
```

Comment:
Names are important for Qt and MacOS: you want to vary the name depending on the platform you are on.  SOmething like:
```
import platform

IS_WINDOWS = (platform.system() == "Windows")

class ExitAction(TaskAction):
    """ A standard 'Quit' or 'Exit' menu action. """

    accelerator = "Alt+F4" if IS_WINDOWS else "Ctrl+Q"
    method = "exit"

    def _name_default(self):
        return (u"Exit" if IS_WINDOWS else u"Quit ") + self.application.name
```
This should behave as expected on most platforms, including not getting two "Exit"/"Quit" menus (On OS X the quit menu will automatically be shifted from the File menu to the correct place).

Reply:
As noted below, this should probably be a `TasksApplicationAction` which calls the application's `exit` method.

Reply:
And, now that I think about it, probably shouldn't even be in _this_ extension, and may not be needed at all.

Reply:
It looks like the exit routine has been overridden in a `TaskWindow` subclass in order to provide a custom message / save function: https://github.com/force-h2020/force-wfmanager/blob/master/force_wfmanager/wfmanager.py#L67

This might require a bit of work to untangle